### PR TITLE
Add a-jazz akc084 keyboard

### PR DESCRIPTION
**Pull request by fenggx-a-jazz**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/a_jazz/akc084/akc084.c
Comment (waffle87) [inline keyboards/a_jazz/akc084/akc084.c:1]: This file should be removed.

File: keyboards/a_jazz/akc084/readme.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# A-JAZZ  AKC084
+
+![](https://github.com/fenggx-a-jazz/documents/blob/main/akc084.jpg?raw=true)
+![](https://github.com/fenggx-a-jazz/documents/blob/main/akc084-2.png?raw=true)
+![](https://github.com/fenggx-a-jazz/documents/blob/main/akc084-2.jpg?raw=true)
```
Comment (waffle87) [inline keyboards/a_jazz/akc084/readme.md:5]: Please rehost these images on Imgur.

File: keyboards/a_jazz/akc084/readme.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+* A customizable 84keys keyboard.
+
+* Keyboard Maintainer: [Feng](https://github.com/fenggx-a-jazz)
+* Hardware Supported: [a-jazz](https://www.a-jazz.com)
+* Hardware Availability: [a-jazz](https://www.a-jazz.com)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make a_jazz/akc084:default
```
Comment (waffle87) [inline keyboards/a_jazz/akc084/readme.md:15]:
```suggestion
    make a_jazz/akc084:default

```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/a_jazz/akc084/readme.md (hunk 1 above)
Comment (fenggx-a-jazz) [reply keyboards/a_jazz/akc084/readme.md:5]: I will remove the images for now. Setup imgur soon.

File: keyboards/a_jazz/akc084/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        _______,   _______,   _______,                          _______,               _______,   _______,   _______,   _______,  _______,   _______
+    ),
+    [3] = LAYOUT(
+        _______,   _______,   _______,    _______,   _______,   _______,    _______,   _______,   _______,   _______,   _______,  _______,   _______,  _______,  _______,  _______,   _______,
+        _______,   _______,   _______,    _______,   _______,   _______,    _______,   _______,   _______,   _______,   _______,   _______,  _______,  _______,  _______,
+        _______,   _______,   _______,    _______,   _______,   _______,    _______,   _______,   _______,   _______,   _______,   _______,  _______,  _______,  _______,
+        _______,   _______,   _______,    _______,   _______,   _______,    _______,   _______,   _______,   _______,   _______,   _______,  _______,  _______,
+        _______,   _______,   _______,    _______,   _______,   _______,    _______,   _______,   _______,   _______,   _______,   _______,  _______,  _______,
+        _______,   _______,   _______,                          _______,               _______,   _______,   _______,   _______,   _______,  _______
```
Comment (waffle87) [inline keyboards/a_jazz/akc084/keymaps/default/keymap.c:59]:
```suggestion
```

File: keyboards/a_jazz/akc084/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [0] =  { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)  },
+    [1] =  { ENCODER_CCW_CW(_______, _______)  },
+    [2] =  { ENCODER_CCW_CW(_______, _______)  },
+    [3] =  { ENCODER_CCW_CW(_______, _______)  }
```
Comment (waffle87) [inline keyboards/a_jazz/akc084/keymaps/default/keymap.c:23]:
```suggestion
    [1] =  { ENCODER_CCW_CW(_______, _______)  }
```

File: keyboards/a_jazz/akc084/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [0] =  { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)  },
+    [1] =  { ENCODER_CCW_CW(_______, _______)  },
+    [2] =  { ENCODER_CCW_CW(_______, _______)  },
+    [3] =  { ENCODER_CCW_CW(_______, _______)  }
```
Comment (waffle87) [inline keyboards/a_jazz/akc084/keymaps/via/keymap.c:23]:
```suggestion
    [1] =  { ENCODER_CCW_CW(_______, _______)  }
```

File: keyboards/a_jazz/akc084/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        _______,   _______,   _______,                          _______,               _______,   _______,   _______,   _______,  _______,   _______
+    ),
+    [3] = LAYOUT(
+        _______,   _______,   _______,    _______,   _______,   _______,    _______,   _______,   _______,   _______,   _______,  _______,   _______,  _______,  _______,  _______,   _______,
+        _______,   _______,   _______,    _______,   _______,   _______,    _______,   _______,   _______,   _______,   _______,   _______,  _______,  _______,  _______,
+        _______,   _______,   _______,    _______,   _______,   _______,    _______,   _______,   _______,   _______,   _______,   _______,  _______,  _______,  _______,
+        _______,   _______,   _______,    _______,   _______,   _______,    _______,   _______,   _______,   _______,   _______,   _______,  _______,  _______,
+        _______,   _______,   _______,    _______,   _______,   _______,    _______,   _______,   _______,   _______,   _______,   _______,  _______,  _______,
+        _______,   _______,   _______,                          _______,               _______,   _______,   _______,   _______,   _______,  _______
```
Comment (waffle87) [inline keyboards/a_jazz/akc084/keymaps/via/keymap.c:59]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/a_jazz/akc084/keyboard.json [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    "matrix_pins": {
+        "cols": ["B0", "B1", "B2", "B3", "B4", "B5", "B6", "B7", "B8", "B9", "B10", "B11", "B12", "B13"],
+        "rows": ["A0", "A1", "A2", "A3", "A4", "A5", "A8"]
+    },
+    "processor": "STM32F103",
+    "usb": {
+        "device_version": "1.0.1",
+        "no_startup_check": true,
+        "pid": "0x0084",
+        "vid": "0x3151",
```
Comment (drashna) [inline keyboards/a_jazz/akc084/keyboard.json:34]: This VID is already in use, and should be changed
https://www.the-sz.com/products/usbid/index.php?v=0x3151&p=&n=

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/a_jazz/akc084/keyboard.json (hunk 7 above)
Comment (fenggx-a-jazz) [reply keyboards/a_jazz/akc084/keyboard.json:34]: Hi, 0x28E9 is the vid of GDdevice.

Comment (fenggx-a-jazz) [reply keyboards/a_jazz/akc084/keyboard.json]: change VID 0X414A

File: keyboards/a_jazz/akc084/readme.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
+# A-JAZZ  AKC084
+
+* A customizable 84keys keyboard.
+![akc084](https://i.imgur.com/381vaD7.png)
+* Keyboard Maintainer: [Feng](https://github.com/fenggx-a-jazz)
+* Hardware Supported: [a-jazz](https://www.a-jazz.com)
+* Hardware Availability: [a-jazz](https://www.a-jazz.com)
```
Comment (waffle87) [inline keyboards/a_jazz/akc084/readme.md:7]: A couple things...
1. Can you please provide a link to this keyboard's store listing page? I'm unable to find anything regarding this keyboard on the internet.
2. Is this a wireless or "tri-mode" keyboard? I am able to find [Ajazz AKC087](https://ajazzstore.com/collections/all/products/ajazz-akc087), which is advertised as a "Bluetooth 5.0/2.4G Wireless & Type-C Wired" keyboard.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/a_jazz/akc084/readme.md (hunk 8 above)
Comment (fenggx-a-jazz) [reply keyboards/a_jazz/akc084/readme.md:8]: https://ajazzstore.com/collections/all/products/ajazz-akp846?variant=43770515587125

Comment (waffle87) [reply keyboards/a_jazz/akc084/readme.md:8]: Thanks!

File: keyboards/a_jazz/akc084/readme.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
+# A-JAZZ  AKC084
+
+A customizable 84keys keyboard
+
+![akc084](https://i.imgur.com/381vaD7.png)
+* Keyboard Maintainer: [Feng](https://github.com/fenggx-a-jazz)
+* Hardware Supported: [a-jazz](https://www.a-jazz.com)
+* Hardware Availability: [a-jazz](https://www.a-jazz.com)
```
Comment (waffle87) [inline keyboards/a_jazz/akc084/readme.md:8]:
```suggestion
* Hardware Availability: [a-jazz]([ajazzstore.com/collections/all/products/ajazz-akp846](https://ajazzstore.com/collections/all/products/ajazz-akp846?variant=43770515587125))
```

**Review by waffle87 — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by armondavani** [comment]
@fenggx-a-jazz Would this profile work for the AJAZZ AKP846? It has 6 rows instead of 5. Thank you

File: keyboards/a_jazz/akc084/readme.md (hunk 9 above)
Comment (armondavani) [reply keyboards/a_jazz/akc084/readme.md]: The linked hardware availability is a 6 row keyboard, but the profile updated shows for a 5 row. This doesn't seem to be the same keyboard.

**Pull request merged**